development/tools/pkg/githubrelease/github.go

```diff
@@ -4,21 +4,45 @@ import (
 	"context"
 	"os"
 
+	"golang.org/x/oauth2"
+
 	"github.com/google/go-github/github"
 	"github.com/kyma-project/test-infra/development/tools/pkg/common"
 )
 
-// GithubAPIWrapper abstracts Github API
-type GithubAPIWrapper struct {
-	Context   context.Context
-	Client    *github.Client
-	RepoOwner string
-	RepoName  string
+// GithubAPI exposes functions to interact with Github releases
+type GithubAPI interface {
+	CreateGithubRelease(ctx context.Context, releaseVersion string, releaseBody string, targetCommit string, isPreRelease bool) (*github.RepositoryRelease, *github.Response, error)
+	UploadFile(ctx context.Context, releaseID int64, artifactName string, artifactFile *os.File) (*github.ReleaseAsset, *github.Response, error)
+}
+
+// githubAPIWrapper implements functions to interact with Github releases
+type githubAPIWrapper struct {
+	githubClient *github.Client
+	repoOwner    string
+	repoName     string
+}
+
+// NewGithubAPIWrapper returns new instance of githubAPIWrapper
+func NewGithubAPIWrapper(ctx context.Context, githubAccessToken, repoOwner, repoName string) GithubAPI {
```

Comment:
If you're returning `GithubAPI` why is the constructor name ends with wrapper?

Reply:
done